src/Sarif.Viewer.VisualStudio/Sarif/Run.extensions.cs

```diff
@@ -9,13 +9,20 @@ namespace Microsoft.Sarif.Viewer.Sarif
 {
     static class RunExtensions
     {
-        public static bool TryGetRule(this Run run, string ruleId, out IRule rule)
+        public static bool TryGetRule(this Run run, string ruleId, string ruleKey, out IRule rule)
         {
             rule = null;
 
             if (run != null && run.Rules != null)
             {
-                rule = run.Rules.Values.FirstOrDefault(r => r.Id == ruleId);
+                if (ruleKey != null)
+                {
+                    rule = run.Rules[ruleKey];
```

Comment:
This method is called `Try`, but it does a direct dictionary lookup, which might throw.